h2o-core/src/main/java/water/Job.java

```diff
@@ -60,6 +61,19 @@ public void setWarnings(final String [] warns){
     }.apply(this);
   }
 
+  public void setResult(final Key<T> result) {
+    String resultClassName = TypeMap.className(_typeid);
+    if (!"hex.generic.GenericModel".equals(resultClassName)) {
+      throw new UnsupportedOperationException("Result key cannot be modified for " + 
+              (_typeid == 0 ? "unknown result type." : "result of type " + resultClassName + "."));
+    }
```

Comment:
It's great to have a safeguard here, but as `_result` is public and not final anymore, the protection is relative.

as search of `_result` read-access shows 52 results, not as large I'd have expected, so… what about making `_result` private and expose a public getter instead? Backward compatibility concerns? Would be safer though.

Reply:
Another approach would be to keep the final modifier and just 
- replace the whole job instance
- use reflection to mutate the final field

I would prefer to keep the typical case the same, don't affect the API - for this very specific need. Maybe I need a new different Job implementation for that.

Reply:
Sure `Key` class is currently `final` but it doesn't appear as a necessity